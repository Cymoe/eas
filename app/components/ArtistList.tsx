import React, { useState, useCallback, useEffect } from 'react';
import {
  View,
  FlatList,
  ActivityIndicator,
  StyleSheet,
  Dimensions,
  Text,
  Image,
} from 'react-native';
import { spotifyService, SpotifyArtist, POPULAR_ARTISTS } from '../../services/spotify';

interface ArtistListProps {
  searchQuery?: string;
  genre?: keyof typeof POPULAR_ARTISTS;
  isRandom?: boolean;
}

export const ArtistList = ({ searchQuery, genre, isRandom }: ArtistListProps) => {
  const [artists, setArtists] = useState<SpotifyArtist[]>([]);
  const [loading, setLoading] = useState(false);
  const [hasMore, setHasMore] = useState(true);
  const [offset, setOffset] = useState(0);
  const [error, setError] = useState<string | null>(null);
  const [isInitialLoad, setIsInitialLoad] = useState(true);

  const loadArtists = useCallback(async () => {
    console.log('loadArtists called', { loading, hasMore, offset, isInitialLoad });
    if (loading || !hasMore) {
      console.log('Skipping load:', { loading, hasMore });
      return;
    }

    try {
      setLoading(true);
      setError(null);

      let result;
      if (searchQuery) {
        console.log('Loading search results:', { searchQuery, offset });
        result = await spotifyService.searchArtistsWithPagination(searchQuery, offset);
      } else if (genre && Object.keys(POPULAR_ARTISTS).includes(genre)) {
        console.log('Loading genre:', { genre, offset });
        result = await spotifyService.getArtistsByGenreWithPagination(
          genre as keyof typeof POPULAR_ARTISTS,
          offset
        );
      } else if (isRandom) {
        console.log('Loading random:', { offset });
        result = await spotifyService.getRandomArtistsWithPagination(offset, 20, artists.map(a => a.id));
      }

      console.log('Load result:', { 
        itemsReceived: result?.items?.length, 
        hasMore: result?.hasMore,
        nextOffset: result?.nextOffset 
      });

      if (!result || result.items.length === 0) {
        setHasMore(false);
      } else {
        setArtists(prev => [...prev, ...result.items]);
        setHasMore(result.hasMore);
        setOffset(result.nextOffset);
      }
    } catch (err) {
      console.error('Load error:', err);
      setError(err instanceof Error ? err.message : 'Failed to load artists');
    } finally {
      setLoading(false);
      setIsInitialLoad(false);
    }
  }, [searchQuery, genre, isRandom, offset, loading, hasMore, artists]);

  useEffect(() => {
    console.log('Input changed, resetting list:', { searchQuery, genre, isRandom });
    setArtists([]);
    setOffset(0);
    setHasMore(true);
    setIsInitialLoad(true);
    loadArtists();
  }, [searchQuery, genre, isRandom]);

  const handleEndReached = useCallback(() => {
    console.log('End reached', { isInitialLoad, loading, hasMore });
    if (!isInitialLoad && !loading && hasMore) {
      loadArtists();
    }
  }, [isInitialLoad, loading, hasMore, loadArtists]);

  const renderArtist = ({ item }: { item: SpotifyArtist }) => {
    const imageUrl = item.images[0]?.url || 'default_image_url';
    
    return (
      <View style={styles.artistCard}>
        <Image
          source={{ uri: imageUrl }}
          style={styles.artistImage}
        />
        <View style={styles.artistInfo}>
          <Text style={styles.artistName}>
            {item.name}
          </Text>
          {item.genres.length > 0 && (
            <Text style={styles.genres}>
              {item.genres.slice(0, 2).join(', ')}
            </Text>
          )}
        </View>
      </View>
    );
  };

  const renderFooter = () => {
    if (!loading) return null;
    return (
      <View style={styles.footer}>
        <ActivityIndicator size="large" color="#FF4B4B" />
      </View>
    );
  };

  const renderEmpty = () => {
    if (loading && isInitialLoad) return null;
    return (
      <View style={styles.emptyContainer}>
        <Text style={styles.emptyText}>No artists found</Text>
      </View>
    );
  };

  return (
    <View style={styles.container}>
      {error ? (
        <Text style={styles.error}>{error}</Text>
      ) : (
        <FlatList
          data={artists}
          renderItem={renderArtist}
          keyExtractor={item => item.id}
          onEndReached={handleEndReached}
          onEndReachedThreshold={0.2}
          onScroll={event => {
            const { layoutMeasurement, contentOffset, contentSize } = event.nativeEvent;
            const paddingToBottom = 20;
            const isCloseToBottom = layoutMeasurement.height + contentOffset.y >= 
              contentSize.height - paddingToBottom;
            
            console.log('Scroll position:', {
              isCloseToBottom,
              currentHeight: layoutMeasurement.height + contentOffset.y,
              totalHeight: contentSize.height,
              remaining: contentSize.height - (layoutMeasurement.height + contentOffset.y)
            });
          }}
          ListFooterComponent={renderFooter}
          ListEmptyComponent={renderEmpty}
          numColumns={2}
          columnWrapperStyle={styles.row}
          contentContainerStyle={[
            styles.list,
            artists.length === 0 && styles.emptyList
          ]}
          showsVerticalScrollIndicator={false}
          removeClippedSubviews={false}
          windowSize={5}
          maxToRenderPerBatch={10}
          updateCellsBatchingPeriod={50}
        />
      )}
    </View>
  );
};

const { width } = Dimensions.get('window');
const CARD_MARGIN = 8;
const CARD_WIDTH = (width - 32 - CARD_MARGIN) / 2;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  list: {
    padding: 16,
  },
  emptyList: {
    flex: 1,
    justifyContent: 'center',
  },
  row: {
    justifyContent: 'space-between',
  },
  artistCard: {
    width: CARD_WIDTH,
    marginBottom: 12,
    borderRadius: 12,
    backgroundColor: 'rgba(255, 255, 255, 0.08)',
    overflow: 'hidden',
  },
  artistImage: {
    width: CARD_WIDTH,
    height: CARD_WIDTH,
  },
  artistInfo: {
    padding: 8,
  },
  artistName: {
    fontSize: 16,
    fontWeight: '500',
    color: '#FFFFFF',
  },
  genres: {
    fontSize: 12,
    marginTop: 4,
    color: 'rgba(255, 255, 255, 0.48)',
  },
  footer: {
    paddingVertical: 12,
  },
  error: {
    padding: 12,
    textAlign: 'center',
    color: '#F41857',
  },
  emptyContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 16,
  },
  emptyText: {
    fontSize: 16,
    color: 'rgba(255, 255, 255, 0.48)',
  },
}); 