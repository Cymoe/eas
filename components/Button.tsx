import { TouchableOpacity, StyleSheet, ViewStyle, TextStyle } from 'react-native';
import { ThemedText } from './ThemedText';

interface ButtonProps {
  variant?: 'primary' | 'secondary' | 'ghost';
  label: string;
  onPress: () => void;
  disabled?: boolean;
  compact?: boolean;
  style?: ViewStyle;
}

export const Button = ({
  variant = 'primary',
  label,
  onPress,
  disabled = false,
  compact = false,
  style,
}: ButtonProps) => {
  const opacity = disabled ? 0.48 : 1;
  const height = compact ? 40 : 48;

  const getButtonStyle = (): ViewStyle => {
    const baseStyle: ViewStyle = {
      height,
      paddingHorizontal: 16,
      borderRadius: 9999, // Fully rounded as per design system
      alignItems: 'center',
      justifyContent: 'center',
      opacity,
    };

    switch (variant) {
      case 'primary':
        return {
          ...baseStyle,
          backgroundColor: '#FF4B4B',
        };
      case 'secondary':
        return {
          ...baseStyle,
          backgroundColor: 'transparent',
          borderWidth: 1,
          borderColor: `rgba(255, 75, 75, ${opacity})`,
        };
      case 'ghost':
        return {
          ...baseStyle,
          backgroundColor: 'transparent',
        };
    }
  };

  const getTextStyle = (): TextStyle => {
    const baseStyle: TextStyle = {
      fontSize: compact ? 14 : 16,
      fontWeight: '600',
    };

    switch (variant) {
      case 'primary':
        return {
          ...baseStyle,
          color: '#FFFFFF',
        };
      case 'secondary':
        return {
          ...baseStyle,
          color: '#FF4B4B',
        };
      case 'ghost':
        return {
          ...baseStyle,
          color: '#A0A0A0',
        };
    }
  };

  return (
    <TouchableOpacity
      style={[getButtonStyle(), style]}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.64} // As per design system hover state
    >
      <ThemedText style={getTextStyle()}>{label}</ThemedText>
    </TouchableOpacity>
  );
}; 