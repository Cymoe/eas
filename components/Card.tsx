import { View, StyleSheet, ViewStyle } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

interface CardProps {
  children: React.ReactNode;
  variant?: 'standard' | 'dialog' | 'menu';
  withGradient?: boolean;
  style?: ViewStyle;
}

export const Card = ({
  children,
  variant = 'standard',
  withGradient = false,
  style,
}: CardProps) => {
  const getCardStyle = (): ViewStyle => {
    const baseStyle: ViewStyle = {
      borderRadius: 12, // Maximum 12px as per design system
      overflow: 'hidden',
    };

    switch (variant) {
      case 'dialog':
        return {
          ...baseStyle,
          padding: 24,
          backgroundColor: '#121212',
        };
      case 'menu':
        return {
          ...baseStyle,
          padding: 12,
          backgroundColor: '#121212',
        };
      default:
        return {
          ...baseStyle,
          padding: 16,
          backgroundColor: 'rgba(255,255,255,0.08)',
        };
    }
  };

  return (
    <View style={[getCardStyle(), style]}>
      {children}
      {withGradient && (
        <LinearGradient
          colors={['rgba(255,75,75,0.08)', 'rgba(30,215,96,0.08)']}
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 1 }}
          style={styles.gradient}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  gradient: {
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    height: '100%',
  },
}); 