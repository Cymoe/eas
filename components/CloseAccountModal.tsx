import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Modal,
  Dimensions,
  ScrollView
} from 'react-native';
import { BlurView } from 'expo-blur';

const { width, height } = Dimensions.get('window');

interface CloseAccountModalProps {
  visible: boolean;
  onClose: () => void;
  onCloseAccount: () => void;
}

export default function CloseAccountModal({ visible, onClose, onCloseAccount }: CloseAccountModalProps) {
  return (
    <Modal
      animationType="fade"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <BlurView 
        intensity={30} 
        tint="dark" 
        style={styles.modalOverlay}
      >
        <View style={styles.modalContainer}>
          <ScrollView style={styles.scrollView} showsVerticalScrollIndicator={false}>
            <Text style={styles.modalTitle}>Closing your account and deleting your data</Text>
            
            <Text style={styles.modalText}>
              To close your <Text style={styles.boldText}>BandMate</Text> Free or Premium 
              account and delete your data permanently from all <Text style={styles.boldText}>BandMate</Text> apps and services, you 
              can use our automated service.
            </Text>
            
            <Text style={styles.modalText}>
              If you're looking cancel your Premium subscription, you can keep all your playlists 
              and saved music by switching to <Text style={styles.boldText}>BandMate</Text> free, our ad-supported service.
            </Text>
            
            <TouchableOpacity 
              style={styles.closeAccountButton} 
              onPress={onCloseAccount}
            >
              <Text style={styles.closeAccountButtonText}>Close my account and delete my data</Text>
            </TouchableOpacity>
            
            <Text style={styles.disclaimerText}>
              After you close your account, we will email you a link which 
              you can use to reactivate it within 30 days. After those 30 
              days, your account cannot be reactivated and the 
              process to delete your data will be initiated. You can 
              always create a new one.
            </Text>
            
            <Text style={styles.disclaimerText}>
              Information you submit will be used to respond to your 
              enquiry, and as otherwise described in our <Text style={styles.linkText}>Privacy Policy</Text>.
            </Text>
            
            <TouchableOpacity onPress={() => {}}>
              <Text style={styles.readMoreText}>Read more</Text>
            </TouchableOpacity>
          </ScrollView>
          
          <TouchableOpacity style={styles.closeButton} onPress={onClose}>
            <Text style={styles.closeButtonText}>âœ•</Text>
          </TouchableOpacity>
        </View>
      </BlurView>
    </Modal>
  );
}

const styles = StyleSheet.create({
  modalOverlay: {
    flex: 1,
    justifyContent: 'flex-end',
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
  },
  modalContainer: {
    backgroundColor: '#1E1E1E',
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    padding: 20,
    paddingBottom: 40,
    width: '100%',
    maxHeight: height * 0.8,
  },
  scrollView: {
    marginBottom: 20,
  },
  modalTitle: {
    fontFamily: 'Abril Fatface',
    fontSize: 28,
    color: '#FFFFFF',
    marginBottom: 16,
  },
  modalText: {
    fontFamily: 'Poppins',
    fontSize: 16,
    color: 'rgba(255, 255, 255, 0.7)',
    marginBottom: 16,
    lineHeight: 24,
  },
  boldText: {
    fontWeight: '600',
    color: '#FFFFFF',
  },
  closeAccountButton: {
    backgroundColor: '#FF3B30',
    borderRadius: 9999,
    paddingVertical: 16,
    paddingHorizontal: 24,
    alignItems: 'center',
    marginVertical: 24,
  },
  closeAccountButtonText: {
    fontFamily: 'Poppins',
    fontSize: 16,
    fontWeight: '500',
    color: '#FFFFFF',
  },
  disclaimerText: {
    fontFamily: 'Poppins',
    fontSize: 14,
    color: 'rgba(255, 255, 255, 0.5)',
    marginBottom: 12,
    lineHeight: 20,
  },
  linkText: {
    color: '#FF3B30',
    textDecorationLine: 'underline',
  },
  readMoreText: {
    fontFamily: 'Poppins',
    fontSize: 14,
    color: '#FF3B30',
    marginTop: 8,
  },
  closeButton: {
    width: 48,
    height: 48,
    borderRadius: 24,
    backgroundColor: 'rgba(255, 255, 255, 0.16)',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
  },
  closeButtonText: {
    color: '#FFFFFF',
    fontSize: 18,
  },
});
