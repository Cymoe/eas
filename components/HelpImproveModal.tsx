import React, { useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Modal,
  TextInput,
  Dimensions
} from 'react-native';
import { BlurView } from 'expo-blur';
import { Ionicons } from '@expo/vector-icons';

const { width, height } = Dimensions.get('window');

interface HelpImproveModalProps {
  visible: boolean;
  onClose: () => void;
  userEmail?: string;
}

export default function HelpImproveModal({ 
  visible, 
  onClose, 
  userEmail = "abc@gmail.com" 
}: HelpImproveModalProps) {
  const [rating, setRating] = useState<number | null>(null);
  const [feedback, setFeedback] = useState('');
  const [charCount, setCharCount] = useState(0);
  const maxCharCount = 500;

  const handleTextChange = (text: string) => {
    if (text.length <= maxCharCount) {
      setFeedback(text);
      setCharCount(text.length);
    }
  };

  const handleRatingSelect = (selectedRating: number) => {
    setRating(selectedRating);
  };

  const handleResetPassword = () => {
    // Logic to reset password would go here
    console.log("Reset password requested");
    // Could show a success message or navigate to a reset screen
  };

  const handleKeepTrying = () => {
    // Logic for "Keep trying" option
    console.log("Keep trying selected");
    onClose();
  };

  const handleReportMusician = () => {
    // Logic to report musician would go here
    console.log("Report musician selected");
    // Could navigate to a report screen
  };

  return (
    <Modal
      animationType="fade"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <BlurView 
        intensity={30} 
        tint="dark" 
        style={styles.modalOverlay}
      >
        <View style={styles.modalContainer}>
          <Text style={styles.modalTitle}>Help us improve BandMate</Text>
          
          {/* Rating Buttons */}
          <View style={styles.ratingContainer}>
            {[1, 2, 3, 4, 5].map((num) => (
              <TouchableOpacity
                key={num}
                style={[
                  styles.ratingButton,
                  rating === num && styles.selectedRating
                ]}
                onPress={() => handleRatingSelect(num)}
              >
                <Text style={[
                  styles.ratingText,
                  rating === num && styles.selectedRatingText
                ]}>
                  {num}
                </Text>
              </TouchableOpacity>
            ))}
          </View>
          
          <Text style={styles.ratingLabel}>
            <Ionicons name="checkmark-circle" size={14} color="rgba(255, 255, 255, 0.48)" /> Agree.
          </Text>
          
          <Text style={styles.feedbackLabel}>
            Let us know how we can improve your experience on BandMate
          </Text>
          
          {/* Feedback Text Input */}
          <TextInput
            style={styles.feedbackInput}
            placeholder="e.g. I have an thought about..."
            placeholderTextColor="rgba(255, 255, 255, 0.3)"
            multiline
            value={feedback}
            onChangeText={handleTextChange}
          />
          
          <Text style={styles.charCounter}>
            <Ionicons name="information-circle" size={14} color="rgba(255, 255, 255, 0.48)" /> Maximum {charCount}/{maxCharCount} characters.
          </Text>
          
          {/* Action Buttons */}
          <TouchableOpacity 
            style={styles.actionButton}
            onPress={handleResetPassword}
          >
            <View style={styles.actionIconContainer}>
              <Ionicons name="mail" size={24} color="#FFFFFF" />
            </View>
            <View style={styles.actionTextContainer}>
              <Text style={styles.actionTitle}>Ask to reset my password</Text>
              <Text style={styles.actionSubtitle}>
                A verification code will be sent {userEmail} to you to reset your password.
              </Text>
            </View>
          </TouchableOpacity>
          
          <TouchableOpacity 
            style={styles.actionButton}
            onPress={handleKeepTrying}
          >
            <View style={styles.actionTextContainer}>
              <Text style={styles.actionTitle}>Keep trying</Text>
              <Text style={styles.actionSubtitle}>
                Try again but with a new password.
              </Text>
            </View>
          </TouchableOpacity>
          
          <TouchableOpacity 
            style={styles.actionButton}
            onPress={handleReportMusician}
          >
            <View style={styles.actionTextContainer}>
              <Text style={styles.actionTitle}>Yes, report this musician</Text>
              <Text style={styles.actionSubtitle}>
                Select a reason why and motivate it.
              </Text>
            </View>
            <Ionicons name="chevron-forward" size={24} color="#FFFFFF" />
          </TouchableOpacity>
          
          <TouchableOpacity style={styles.closeButton} onPress={onClose}>
            <Text style={styles.closeButtonText}>âœ•</Text>
          </TouchableOpacity>
        </View>
      </BlurView>
    </Modal>
  );
}

const styles = StyleSheet.create({
  modalOverlay: {
    flex: 1,
    justifyContent: 'flex-end',
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
  },
  modalContainer: {
    backgroundColor: '#1E1E1E',
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    padding: 20,
    paddingBottom: 40,
    width: '100%',
    maxHeight: height * 0.9,
  },
  modalTitle: {
    fontFamily: 'Abril Fatface',
    fontSize: 24,
    color: '#FFFFFF',
    marginBottom: 16,
  },
  ratingContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 8,
  },
  ratingButton: {
    width: 60,
    height: 44,
    borderRadius: 12,
    backgroundColor: '#262626',
    alignItems: 'center',
    justifyContent: 'center',
  },
  selectedRating: {
    backgroundColor: '#FFFFFF',
  },
  ratingText: {
    fontFamily: 'Poppins',
    fontSize: 16,
    fontWeight: '500',
    color: '#FFFFFF',
  },
  selectedRatingText: {
    color: '#000000',
  },
  ratingLabel: {
    fontFamily: 'Poppins',
    fontSize: 12,
    color: 'rgba(255, 255, 255, 0.48)',
    marginBottom: 16,
  },
  feedbackLabel: {
    fontFamily: 'Poppins',
    fontSize: 16,
    color: '#FFFFFF',
    marginBottom: 12,
  },
  feedbackInput: {
    backgroundColor: '#262626',
    borderRadius: 12,
    padding: 16,
    height: 120,
    color: '#FFFFFF',
    fontFamily: 'Poppins',
    fontSize: 14,
    textAlignVertical: 'top',
    marginBottom: 8,
  },
  charCounter: {
    fontFamily: 'Poppins',
    fontSize: 12,
    color: 'rgba(255, 255, 255, 0.48)',
    marginBottom: 24,
  },
  actionButton: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#262626',
    borderRadius: 12,
    padding: 16,
    marginBottom: 12,
  },
  actionIconContainer: {
    marginRight: 12,
  },
  actionTextContainer: {
    flex: 1,
  },
  actionTitle: {
    fontFamily: 'Poppins',
    fontSize: 16,
    fontWeight: '500',
    color: '#FFFFFF',
    marginBottom: 4,
  },
  actionSubtitle: {
    fontFamily: 'Poppins',
    fontSize: 14,
    color: 'rgba(255, 255, 255, 0.48)',
  },
  closeButton: {
    width: 48,
    height: 48,
    borderRadius: 24,
    backgroundColor: 'rgba(255, 255, 255, 0.16)',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
    marginTop: 16,
  },
  closeButtonText: {
    color: '#FFFFFF',
    fontSize: 18,
  },
});
