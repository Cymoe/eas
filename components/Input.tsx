import { View, TextInput, StyleSheet, ViewStyle, TextStyle } from 'react-native';
import { ThemedText } from './ThemedText';

interface InputProps {
  label: string;
  value: string;
  onChangeText: (text: string) => void;
  placeholder?: string;
  error?: string;
  success?: string;
  style?: ViewStyle;
}

export const Input = ({
  label,
  value,
  onChangeText,
  placeholder,
  error,
  success,
  style,
}: InputProps) => {
  const getHelperColor = () => {
    if (error) return '#F41857';
    if (success) return '#6BFF90';
    return 'rgba(255,255,255,0.48)';
  };

  const getHelperText = () => {
    if (error) return error;
    if (success) return success;
    return '';
  };

  return (
    <View style={[styles.container, style]}>
      <ThemedText style={styles.label}>{label}</ThemedText>
      <View
        style={[
          styles.inputContainer,
          error && styles.errorInput,
          success && styles.successInput,
        ]}
      >
        <TextInput
          value={value}
          onChangeText={onChangeText}
          placeholder={placeholder}
          placeholderTextColor="rgba(255,255,255,0.48)"
          style={[
            styles.input,
            { color: value ? '#FFFFFF' : 'rgba(255,255,255,0.48)' },
          ]}
        />
      </View>
      {(error || success) && (
        <ThemedText style={[styles.helperText, { color: getHelperColor() }]}>
          {getHelperText()}
        </ThemedText>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    gap: 8,
  },
  label: {
    fontSize: 14,
    color: 'rgba(255,255,255,0.48)',
    textTransform: 'uppercase',
    fontWeight: '500',
  },
  inputContainer: {
    height: 40,
    backgroundColor: 'rgba(255,255,255,0.08)',
    borderRadius: 12,
    paddingHorizontal: 12,
    justifyContent: 'center',
  },
  input: {
    fontSize: 16,
    height: '100%',
    padding: 0, // Remove default padding
  },
  errorInput: {
    backgroundColor: 'rgba(244,24,87,0.08)',
  },
  successInput: {
    backgroundColor: 'rgba(107,255,144,0.08)',
  },
  helperText: {
    fontSize: 12,
    marginTop: 4,
  },
}); 