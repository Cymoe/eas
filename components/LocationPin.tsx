import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface LocationPinProps {
  size?: number;
  color?: string;
}

/**
 * LocationPin component that follows the WindSurf design system
 * Used to display location markers in the app
 */
const LocationPin: React.FC<LocationPinProps> = ({ 
  size = 24, 
  color = '#FF3B30' // Accent color from WindSurf design system
}) => {
  return (
    <View style={styles.container}>
      <Ionicons 
        name="location" 
        size={size} 
        color={color} 
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  }
});

export default LocationPin;
