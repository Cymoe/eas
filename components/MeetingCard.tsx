import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { router } from 'expo-router';

interface MeetingCardProps {
  date?: Date;
  time?: string;
  location?: string;
  city?: string;
  name?: string;
}

const MeetingCard: React.FC<MeetingCardProps> = ({
  date = new Date(),
  time = '--',
  location = '--',
  city = '--',
  name = '--'
}) => {
  const month = date.toLocaleString('default', { month: 'short' });
  const day = date.getDate();

  return (
    <View style={styles.meetingCardWrapper}>
      <View style={styles.meetingCard}>
        <View style={styles.meetingCardHeader}>
          <Text style={styles.meetingCardMonth}>{month}</Text>
          <Text style={styles.meetingCardDay}>{day}</Text>
        </View>
        
        <View style={styles.meetingCardContent}>
          <Text style={styles.meetingCardName}>{name}</Text>
          <View style={styles.meetingCardDetail}>
            <Text style={styles.meetingCardDetailText}>{time}</Text>
          </View>
          <View style={styles.meetingCardDetail}>
            <Text style={styles.meetingCardDetailText}>{`${location}, ${city}`}</Text>
          </View>
        </View>
        
        <TouchableOpacity 
          style={styles.meetingCardEditButton}
          onPress={() => router.push('/meeting-summary')}
        >
          <Text style={styles.editButtonText}>Edit</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  meetingCardWrapper: {
    alignSelf: 'flex-end',
    marginVertical: 8,
    marginHorizontal: 16,
    maxWidth: '80%',
  },
  meetingCard: {
    flexDirection: 'row',
    backgroundColor: '#1F1F1F',
    borderRadius: 12,
    overflow: 'hidden',
    borderWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.1)',
  },
  meetingCardHeader: {
    width: 60,
    backgroundColor: '#121212',
    padding: 8,
    alignItems: 'center',
    justifyContent: 'center',
  },
  meetingCardMonth: {
    fontSize: 14,
    fontWeight: '500',
    color: '#FFFFFF',
    textTransform: 'uppercase',
  },
  meetingCardDay: {
    fontSize: 24,
    fontWeight: '600',
    color: '#FFFFFF',
  },
  meetingCardContent: {
    flex: 1,
    padding: 12,
  },
  meetingCardName: {
    fontSize: 16,
    fontWeight: '600',
    color: '#FFFFFF',
    marginBottom: 8,
  },
  meetingCardDetail: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 4,
  },
  meetingCardDetailText: {
    fontSize: 12,
    color: 'rgba(255, 255, 255, 0.7)',
    marginLeft: 4,
  },
  meetingCardEditButton: {
    width: 40,
    height: 40,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-end',
  },
  editButtonText: {
    color: '#FFFFFF',
    fontSize: 14,
  },
});

export default MeetingCard; 