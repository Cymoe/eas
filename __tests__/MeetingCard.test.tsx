import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { router } from 'expo-router';
import { MeetingCard } from '../app/components/MeetingCard';

// Mock expo-router
jest.mock('expo-router', () => ({
  router: {
    push: jest.fn(),
  },
}));

describe('MeetingCard', () => {
  const mockMeetingData = {
    date: new Date('2024-01-18'),
    time: '6:00PM',
    location: 'Honda Center',
    city: 'Geneva',
    name: 'Anto'
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all props', () => {
    const { getByText } = render(
      <MeetingCard
        date={mockMeetingData.date}
        time={mockMeetingData.time}
        location={mockMeetingData.location}
        city={mockMeetingData.city}
        name={mockMeetingData.name}
      />
    );

    // Check if all the meeting details are rendered
    expect(getByText('Jan')).toBeTruthy();
    expect(getByText('18')).toBeTruthy();
    expect(getByText('Anto')).toBeTruthy();
    expect(getByText('6:00PM')).toBeTruthy();
    expect(getByText('Honda Center, Geneva')).toBeTruthy();
  });

  it('navigates to meeting-summary when edit button is pressed', () => {
    const { getByText } = render(
      <MeetingCard
        date={mockMeetingData.date}
        time={mockMeetingData.time}
        location={mockMeetingData.location}
        city={mockMeetingData.city}
        name={mockMeetingData.name}
      />
    );

    // Press the edit button
    fireEvent.press(getByText('Edit'));

    // Check if router.push was called with correct params
    expect(router.push).toHaveBeenCalledWith('/meeting-summary');
  });

  it('handles undefined props gracefully', () => {
    const { getByText } = render(<MeetingCard />);

    // Should show placeholder or default values
    expect(getByText('--')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <MeetingCard
        date={mockMeetingData.date}
        time={mockMeetingData.time}
        location={mockMeetingData.location}
        city={mockMeetingData.city}
        name={mockMeetingData.name}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
}); 