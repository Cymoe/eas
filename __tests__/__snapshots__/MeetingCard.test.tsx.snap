// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingCard matches snapshot 1`] = `
<View
  style={
    {
      "alignSelf": "flex-end",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "maxWidth": "80%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#1F1F1F",
        "borderColor": "rgba(255, 255, 255, 0.1)",
        "borderRadius": 12,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#121212",
          "justifyContent": "center",
          "padding": 8,
          "width": 60,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "500",
            "textTransform": "uppercase",
          }
        }
      >
        Jan
      </Text>
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 24,
            "fontWeight": "600",
          }
        }
      >
        18
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "600",
            "marginBottom": 8,
          }
        }
      >
        Anto
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 12,
              "marginLeft": 4,
            }
          }
        >
          6:00PM
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 12,
              "marginLeft": 4,
            }
          }
        >
          Honda Center, Geneva
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 14,
          }
        }
      >
        Edit
      </Text>
    </View>
  </View>
</View>
`;
